assets/src/components/selection/event-select/index.js

```diff
@@ -3,77 +3,79 @@
+import { __ } from '@wordpress/i18n';
+import { withSelect } from '@wordpress/data';
 
-const nowDateAndTime = moment();
-
-const buildEventOptions = ( events ) => {
-	reduce( events, function( options, event ) {
-		options.push(
-			{
-				label: event.EVT_name,
-				value: event.EVT_ID,
-			},
-		);
-		return options;
-	}, [] );
-};
+/**
+ * Internal dependencies
+ */
+import { buildEventOptions } from './build-event-options';
 
-export class EventSelect extends Component {
-	render() {
-		const {
-			events = [],
-			onEventSelect,
-			selectLabel = __( 'Select Event' ),
-			selectedEventId,
-			isLoading = true,
-		} = this.props;
-		if ( isLoading || isEmpty( events ) ) {
-			return <Placeholder key="placeholder"
-				icon="calendar"
-				label={ __( 'EventSelect' ) }
-			>
-				{ isLoading ?
-					<Spinner /> :
-					__(
-						'There are no events to select from. You need to create an event first.',
-						'event_espresso'
-					)
-				}
-			</Placeholder>;
-		}
+const nowDateAndTime = moment();
 
-		return <SelectControl
-			label={ selectLabel }
-			value={ selectedEventId }
-			options={ buildEventOptions( events ) }
-			onChange={ ( value ) => onEventSelect( value ) }
-		/>;
+/**
+ * EventSelect component.
+ * A react component for an event selector.
+ *
+ * @param {Array} events  An empty array or array of Event Entities. See
+ *                          prop-types for shape.
+ * @param {function} onEventSelect  The callback on selection of event.
+ * @param {string} selectLabel The label for the select input.
+ * @param {number} selectedEventId  If provided, the id of the event to
+ *   pre-select.
+ * @param {boolean} isLoading  Whether or not the selector should start in a
+ *                               loading state
+ * @return {Function}  A pure component function.
+ * @constructor
+ */
+export const EventSelect = ( {
+	events,
+	onEventSelect,
+	selectLabel,
+	selectedEventId,
+	isLoading,
+} ) => {
+	if ( isLoading || isEmpty( events ) ) {
+		return <Placeholder key="placeholder"
+			icon="calendar"
+			label={ __( 'EventSelect', 'event_espresso' ) }
+		>
+			{ isLoading ?
+				<Spinner /> :
+				__(
+					'There are no events to select from. You need to create an event first.',
+					'event_espresso',
+				)
+			}
+		</Placeholder>;
 	}
-}
+
+	return <SelectControl
+		label={ selectLabel }
+		value={ selectedEventId }
+		options={ buildEventOptions( events ) }
+		onChange={ ( value ) => onEventSelect( value ) }
+	/>;
+};
 
 /**
  * @todo some of these proptypes are likely reusable in various place so we may
  * want to consider extracting them into a separate file/object that can be
  * included as needed.
- * @type {{events: *, onEventSelect, selectLabel: *, selectedEventId: *,
- *   isLoading: *, attributes: {limit: *, orderBy: *, order: *, showExpired: *,
- *   categorySlug: *, month: *}}}
  */
 EventSelect.propTypes = {
-	events: PropTypes.shape( {
+	events: PropTypes.arrayOf( PropTypes.shape( {
```

Comment:
The original implementation of this was incorrect, so this fixes that (covered by tests as well).